bio/rsem/calculate-expression/wrapper.py

```
@@ -60,15 +60,22 @@
         "output.isoforms_results file name malformed "
         "(rsem will append .isoforms.results suffix)"
     )
-
-reference_prefix = os.path.splitext(snakemake.input.reference[0])[0]
+    
+# BUG input_string is 'r' given the input but is should be the reference base path?
+# subprocess.CalledProcessError: Command 'set -euo pipefail;  rsem-calculate-expression --num-threads 24 --estimate-rspd --calc-ci --strandedness reverse --time --paired-end --alignments results/star/D-1/Aligned.toTranscriptome.out.bam r results/rsem/D-1/D-1  > logs/rsem/calculate_expression/D-1.log 2>&1' returned non-zero exit status 255.
+reference_path = Path(snakemake.input.reference[0])
```

Ruff (0.8.2)

Undefined name `snakemake`

(F821)